app/script/conversation/ConversationRepository.js

```diff
@@ -2735,7 +2748,57 @@ z.conversation.ConversationRepository = class ConversationRepository {
       });
   }
 
-  _handledConversationEvent(entityObject = {}, eventSource, previouslyArchived) {
+  /**
+   * calls the feature specific event handler on the current event being handled
+   *
+   * @private
+   * @param {Conversation} conversationEntity - Conversation targeted by the event
+   * @param {Object} eventJson - JSON data of the event
+   * @param {z.event.EventRepository.SOURCE} eventSource - Source of event
+   * @returns {Promise} Resolves when all the handlers have done their job
+   */
+  _triggerFeatureEventHandlers(conversationEntity, eventJson, eventSource) {
+    const conversationEventHandlers = [this.ephemeralHandler, this.stateHandler];
+    const handlePromises = conversationEventHandlers.map(handler =>
+      handler.handleConversationEvent(conversationEntity, eventJson, eventSource)
+    );
+    return Promise.all(handlePromises).then(() => conversationEntity);
+  }
+
+  /**
+   * Add relevant events to the conversation entity in order for it to be displayed on the UI.
+   * Does nothing if the event type doesn't need to be persisted in memory
+   *
+   * @private
+   * @param {Conversation} conversationEntity - Conversation to add the event to
+   * @param {Object} eventJson - JSON data of the event
+   * @returns {Promise} Resolves when event was handled
+   */
+  _handleEventPersistence(conversationEntity, eventJson) {
```

Comment:
Let's rename this. Persistence is not really a good choice as we aren't storing it in the database here.